web/src/buddy_data.ts

```diff
@@ -300,8 +301,18 @@ function maybe_shrink_list(user_ids: number[], user_filter_text: string): number
 
     // We want to always show PM recipients even if they're inactive.
     const pm_ids_set = narrow_state.pm_ids_set();
+    // We want to show all channel subscribers if we have fewer than 50 of them.
+    const filter_by_stream_id =
+        narrow_state.stream_id() &&
+        peer_data.get_subscribers(narrow_state.stream_id()!).length <= 50;
     user_ids = user_ids.filter(
-        (user_id) => user_is_recently_active(user_id) || user_matches_narrow(user_id, pm_ids_set),
+        (user_id) =>
+            user_is_recently_active(user_id) ||
+            user_matches_narrow(
+                user_id,
+                pm_ids_set,
+                filter_by_stream_id ? narrow_state.stream_id() : undefined,
+            ),
     );
```

Comment:
Is it easy to extend the `buddy_data` tests to verify this logic? One option is doing an `export let` for the cutoff so that test code can override the number.

Also, can you use `get_subscriber_count` instead?

Reply:
Used `get_subscriber_count` and added a test.